web_widget_uom/static/src/js/widget.js

```diff
@@ -0,0 +1,41 @@
+/* Global jscolor */
+odoo.define("web.web_widget_uom", function (require) {
+    "use strict";
+
+    var basic_fields = require("web.basic_fields");
+    var field_registry = require("web.field_registry");
+
+    var FieldUoM = basic_fields.FieldFloat.extend({
+        willStart: function () {
+            return Promise.all([
+                this._getDecimalPlaces(),
+                this._super.apply(this, arguments),
+            ]);
+        },
+        _getDecimalPlaces: function () {
+            var self = this;
+            var uomField =
+                this.nodeOptions.uom_field || this.field.uom_field || "uom_id";
+            var uomID = this.record.data[uomField] && this.record.data[uomField].res_id;
+
+            var qtyField = this.nodeOptions.qty_field || "product_uom_qty";
+            var quantity = this.record.data[qtyField] || 0.0;
+
+            return this._rpc({
+                model: "uom.uom",
```

Comment:
AFAIS this is going to issue a call for each widget and each UoM every time.
I suggest to add some caching by uom.id. If you cannot do it now, add a TODO to the roadmap.

Reply:
I could add cache, any hint on best paretics on how to do it?

Reply:
> When you push new changes please rewrite the main commit by removing the version number, thanks 🙏

Do you mean reword `[14.0][ADD] web_widget_uom` to `[ADD] web_widget_uom`?

Reply:
@simahawk I think this was the last comment standing... could you please help? Thanks!

Reply:
@simahawk roadmap added. Since there are many approving reviews, can we merge? 

Thanks!